spec/datetimeformat.html

```diff
@@ -123,15 +123,33 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
             1. Throw a *RangeError* exception.
           1. Let _timeZone_ be CanonicalizeTimeZoneName(_timeZone_).
         1. Set _dateTimeFormat_.[[TimeZone]] to _timeZone_.
-        1. Let _opt_ be a new Record.
+        1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
+        1. Let _formatOptions_ be a new Record.
+        1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
+        1. Let _hc_ be _dateTimeFormat_.[[HourCycle]].
+        1. If _hc_ is *null*, then
+          1. Set _hc_ to _hcDefault_.
+        1. If _hour12_ is not *undefined*, then
+          1. If _hour12_ is *true*, then
+            1. If _hcDefault_ is *"h11"* or *"h23"*, then
+              1. Set _hc_ to *"h11"*.
+            1. Else,
+              1. Set _hc_ to *"h12"*.
+          1. Else,
+            1. Assert: _hour12_ is *false*.
+            1. If _hcDefault_ is *"h11"* or *"h23"*, then
+              1. Set _hc_ to *"h23"*.
+            1. Else,
+              1. Set _hc_ to *"h24"*.
```

Comment:
This can be simplified as in #602 (which I just created forgetting that it overlaps with this PR).
```suggestion
        1. If _hour12_ is *true*, then
          1. If _hcDefault_ is *"h11"* or *"h23"*, then
            1. Set _hc_ to *"h11"*.
          1. Else,
            1. Set _hc_ to *"h12"*.
        1. Else if _hour12_ is *false*, then
          1. If _hcDefault_ is *"h11"* or *"h23"*, then
            1. Set _hc_ to *"h23"*.
          1. Else,
            1. Set _hc_ to *"h24"*.
        1. Else,
          1. Assert: _hour12_ is *undefined*.
```